internal/juju/models.go

```diff
@@ -119,7 +127,12 @@ func (c *modelsClient) Create(name string, controller string, cloudList []interf
 }
 
 func (c *modelsClient) Read(uuid string) (*string, *params.ModelInfo, error) {
-	client := modelmanager.NewClient(c.conn)
+	conn, err := c.cf.GetConnection(nil)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
Now that we are creating single-shot connections, should there also be a `defer conn.Close()`?